plugins/inputs/kinesis_consumer/README.md

```diff
@@ -54,6 +54,15 @@ and creates metrics using one of the supported [input data formats][].
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
   data_format = "influx"
 
+  ##
+  ## Whether or not to uncompress the data from kinesis
+  ## If you are processing a cloudwatch logs kinesis stream then set this to true
+  ## as AWS compresses cloudwatch log data before it is sent to kinesis (aws
+  ## also base64 encodes the zip byte data before pushing to the stream.  The base64 decoding
+  ## is done automatically by the golang sdk, as data is read from kinesis)
+  ##
+  # decompress = false
```

Comment:
Does it make sense to allow to specify the compression algorithm in order to be more flexible? What if they decide to use `snappy` or similar in the future?

Reply:
> Code looks good, but it might be beneficial to allow the user to specify the decompression/compression algorithm (i.e. `"gzip"` for now) to be future proof. What do you think?

Sounds sensible.  How's something like the following sound:

`decompress: "gzip"`

snappy could requested like the following, i.e:

`decompress: "snappy"`

I can look at adding in the support for snappy at the same time (would be an extra dependency I think).

Reply:
Sounds exactly like what I had in mind. :-) You don't need to add snappy now but I like to be prepared. :-) So only having `"gzip"` or `""` is fine for me. If we hit other compression types we can also add later. But in case you are already aware of a use-case, you can of course add it in one go.

Reply:
Oh and please use `internal.choice` for testing the valid values... ;-)

Reply:
Let me know how you feel about this change to `decompress_with`.  I've added `zlib` as an option; but left out the snappy for now (extra dependency required to do the snappy; whereas zlib is standard in golang).   I could always look at sending an additional PR for snappy.  I added a test for the gzip/zlib/none options.

Happy to make changes.  Thanks for the help.

Reply:
I've tried doing a !retry-failed; but it did not seem to make the CircleCI happier.   Unfortunately I can't see what the output of the circle ci tests are; would you be able to provide me the details?   Locally I'm unable to see failures related to the PR.  (The permissions CircleCI is wanting are very permissive; and is wanting access to all of my repos; private and public).